channel.go

```diff
@@ -621,6 +628,9 @@ func (ch *Channel) getMinConnectionState() connectionState {
 func (ch *Channel) connectionCloseStateChange(c *Connection) {
 	ch.removeClosedConn(c)
 	if peer, ok := ch.rootPeers().Get(c.remotePeerInfo.HostPort); ok {
+		if ch.onPeerStatusChanged != nil {
+			ch.onPeerStatusChanged(peer)
```

Comment:
This is only called on `connectionCloseStateChange` -- we'll also want to deal with new incoming connections.